tests/common/filepath.cpp

```diff
@@ -0,0 +1,188 @@
+	EXPECT_STREQ(Common::FilePath::getExtension("file"             ).c_str(), "");
+	EXPECT_STREQ(Common::FilePath::getExtension("/path/to/"        ).c_str(), "");
+}
+
+GTEST_TEST_F(FilePath, getDirectory) {
+	EXPECT_STREQ(Common::FilePath::getDirectory("/path/to/file.ext").c_str(), "/path/to");
+	EXPECT_STREQ(Common::FilePath::getDirectory("path/to/file.ext" ).c_str(), "path/to");
+	EXPECT_STREQ(Common::FilePath::getDirectory("file.ext"         ).c_str(), "");
+	EXPECT_STREQ(Common::FilePath::getDirectory("/path/to/"        ).c_str(), "/path/to");
+}
+
+GTEST_TEST_F(FilePath, changeExtension) {
+	EXPECT_STREQ(Common::FilePath::changeExtension("/path/file.ext", ".new"  ).c_str(), "/path/file.new");
+	EXPECT_STREQ(Common::FilePath::changeExtension("/path/file.ext", ".new.m").c_str(), "/path/file.new.m");
+	EXPECT_STREQ(Common::FilePath::changeExtension("/path/file.ext", "new"   ).c_str(), "/path/file.new");
+	EXPECT_STREQ(Common::FilePath::changeExtension("/path/file.ext", ""      ).c_str(), "/path/file");
+	EXPECT_STREQ(Common::FilePath::changeExtension("file.ext.ext"  , ".new"  ).c_str(), "file.ext.new");
+	EXPECT_STREQ(Common::FilePath::changeExtension("file.ext.ext"  , ".new.m").c_str(), "file.ext.new.m");
+	EXPECT_STREQ(Common::FilePath::changeExtension("file.ext.ext"  , "new"   ).c_str(), "file.ext.new");
+	EXPECT_STREQ(Common::FilePath::changeExtension("file.ext.ext"  , ""      ).c_str(), "file.ext");
+	EXPECT_STREQ(Common::FilePath::changeExtension("file."         , ".new"  ).c_str(), "file.new");
+	EXPECT_STREQ(Common::FilePath::changeExtension("file."         , ".new.m").c_str(), "file.new.m");
+	EXPECT_STREQ(Common::FilePath::changeExtension("file."         , "new"   ).c_str(), "file.new");
+	EXPECT_STREQ(Common::FilePath::changeExtension("file."         , ""      ).c_str(), "file");
+	EXPECT_STREQ(Common::FilePath::changeExtension("file"          , ".new"  ).c_str(), "file.new");
+	EXPECT_STREQ(Common::FilePath::changeExtension("file"          , ".new.m").c_str(), "file.new.m");
+	EXPECT_STREQ(Common::FilePath::changeExtension("file"          , "new"   ).c_str(), "file.new");
+	EXPECT_STREQ(Common::FilePath::changeExtension("file"          , ""      ).c_str(), "file");
+	EXPECT_STREQ(Common::FilePath::changeExtension("/path/"        , ".new"  ).c_str(), "/path/.new");
+	EXPECT_STREQ(Common::FilePath::changeExtension("/path/"        , ".new.m").c_str(), "/path/.new.m");
+	EXPECT_STREQ(Common::FilePath::changeExtension("/path/"        , "new"   ).c_str(), "/path/.new");
+	EXPECT_STREQ(Common::FilePath::changeExtension("/path/"        , ""      ).c_str(), "/path/");
+}
+
+GTEST_TEST_F(FilePath, relativize) {
+	EXPECT_STREQ(Common::FilePath::relativize("/path/to", "/path/to/file.ext").c_str(), "file.ext");
+	EXPECT_STREQ(Common::FilePath::relativize("/other"  , "/path/to/file.ext").c_str(), "");
+
+	/* Common::FilePath::relativize() only works with normalized paths...
+	 * These here are common failure modes.
```

Comment:
I don't like the behavior of `relativize` only working with normalized paths -- can't it just do the normalization itself?

(Also, `makeRelative` would be a nicer name, IMO)